packages/types/src/config/config.ts

```diff
@@ -0,0 +1,181 @@
+// Copyright 2021-2022 Prosopo (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { LogLevel } from '@prosopo/common'
+import { z } from 'zod'
+
+export enum DatabaseTypes {
+    mongo = 'mongo',
```

Comment:
I think enums like this can be presented differently to avoid the `name = str` pattern, which can lead to accidental inconsistencies, plus we have to type everything twice.

I'm thinking something along the lines of `type DatabaseTypes = 'mongo' | 'mongoMemory';`, thoughts? afaik this is recommended ahead of enum usage in typescript. Given we're not mapping an enum name to a _different_ string key, the `type` union makes more sense imo

Reply:
My IDE auto completes enums like this for me.

I've read [this](https://stackoverflow.com/a/76485953) and it seems like `z.enum` is the way to go. I didn't realise there were issues with native enums.

The main thing I want to avoid is having to write strings everywhere like `'mongo'`, as these aren't picked up when refactoring.